### DISCUSSION TITLE
New License Key Not Working

### DISCUSSION THREAD

**Discussion by MAK1023** [noah-nuebling/mac-mouse-fix#495] [Questions]
Hey everyone,

I've become heavily reliant on Mac Mouse Fix to be productive on my Mac. I absolutely loved it until the beta 6 came out. My free trial was up, so it told me to update and buy a license key. I did that and then I realized that the developer got rid of most of the features that I used every single time I interact with my mouse. MMF is now completely useless to me. I tried deleting beta 6 and reinstalling beta 5, but it won't accept my license key. Desperately, I purchased a second key hoping this would fix this, but it didn't. Any ideas what I can do? I appreciate any help.

Thanks

**Comment by noah-nuebling** [comment]
Hey @MAK1023, sorry for the trouble!

I’ll send you a link for a free license that works with Beta 5. 

Which features are you missing exactly in Beta 6? It should have pretty much the same features as Beta 5. If you use the “Restore Defaults…”  Button that might help you to restore the same functionality you were used to in Beta 5.

Thanks and I hope that helps a little!

**Reply by MAK1023** [reply]
Hey Noah,

I’m no longer able to “click and scroll” to move between spaces, which was HUGE for me. The way I have my mouse’s DPI set to work with my workflow on a dual ultrawide monitor setup, the “click and drag” to move between spaces is unusable, so “click and scroll” was incredible. Also, the directions of “click and drag” to move between spaces/to pan have been reversed from the scroll wheel direction. Previously, it felt very natural to be able to scroll my wheel down to make the page go up, while “click and drag” to pan in the direction that I’m moving my mouse. Now that it’s been reversed, either the scrolling is reversed or the panning is reversed depending on the scroll setting I choose, but neither feels right. 

These were the 2 very most important features that I used every time I interacted with my mouse and absolutely loved MMF for because it changed the game for me. Is there any way to bring “click and scroll” to move between spaces back? As for the scroll/pan directions, I think I saw in the notes that the directionality was changed because of trackpad. Is it possible to separate those 2 (scroll direction vs pan direction) in the options list so that users who use a mouse vs users who use a trackpad choose what works best for them? 

Your work is incredible and MMF is an absolute game-changer like no other for Mac. I appreciate your time!

**Reply by noah-nuebling** [reply]
Thank you so much for the compliments! 
I'll probably be bringing back Click and Scroll to move between Spaces in the next Beta since you like it so much more than Click and Drag.

The Click and Drag direction is supposed to follow the Trackpad direction in Beta 6. If you go to System Settings > Trackpad and toggle the Natural Scrolling option, that should also affect the Click and Drag in Mac Mouse Fix.

I've been receiving a lot of complaints about this though, so there might be some bugs with that. 

My plan is to disable the inversion in the next MMF version and eventually add a toggle for the option in MMF instead of trying to have it follow the option inside System Settings > Trackpad.

Hope that helps a little! Let me know if you have any more questions.

**Reply by noah-nuebling** [reply]
> The way I have my mouse’s DPI set to work with my workflow on a dual ultrawide monitor setup, the “click and drag” to move between spaces is unusable, so “click and scroll” was incredible.

Hey again, MAK1023! The Click and Scroll to move between Spaces feature will probably return in the next release. I'm currently working on that. 

But I think a better solution might be to have the Click and Drag feature work properly even on your ultrawide setup, because that would make the Click and Scroll to move between Spaces feature redundant.

If you have time, I would really appreciate it if you could give me more info on why the Click and Drag feature doesn't work properly on your setup. Maybe you could share a screen recording or something.

Thanks! :)
